dart/example/example.dart

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 Dan Field. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import './monster_my_game.sample_generated.dart' as myGame;
+
+// Example how to use FlatBuffers to create and read binary buffers.
+
+void main() {
+  // Create the builder here so we can use it for both weapons and equipped
+  // the actual data will only be written to the buffer once.
+  var axe = new myGame.WeaponBuilder(name: 'Axe', damage: 5);
```

Comment:
The user could still use the `Builder` class more directly, the way the generated code does.

The "builder helpers" do allocate references to the data that will be held until the helper goes out of scope/is garbage collected.

In a lower level language, this would definitely be a concern and more important to avoid - I'm just not as convinced it makes as much sense to avoid in Dart.  The scale at which this would become a problem in Dart will likely already be a problem for other reasons (in other words, I don't see Dart being the best candidate for processing very large FlatBuffers data, even if it wasn't making any unnecessary allocations).  I also am not convinced that the runtime wouldn't decide to make allocations behind our backs anyway (I've certainly seen that in other garbage collected environments), even though we've made the code more difficult to use by trying to avoid any unnecessary allocations. 

I've basically followed the design that was done in the original SDK work, although in this case I think that it's worth keeping both for ease of use and low likelihood of performance or usability improvements.

Reply:
I agree that it is not likely to be an issue for Dart, but it is more of a style issue. FlatBuffers is about creating and reading buffers in-place, and is about efficiency. Even in Python (probably the slowest language of all) we adhere to this pattern.

Reply:
I'm almost set on this - I've renamed the existing builders to "ObjectBuilder", because I think the syntax/usage will feel more natural and is easier to use for people who don't mind the additional allocations.  But I'm also generating builders closer to what you have in the other languages